android/jni/com/mapswithme/maps/bookmarks/data/Bookmark.cpp

```diff
@@ -81,4 +81,13 @@ Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeEncode2Ge0Url(
 {
   return jni::ToJavaString(env, frm()->CodeGe0url(getBookmark(cat, bmk), addName));
 }
+
+JNIEXPORT jstring JNICALL
+Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeGetAddress(
+     JNIEnv * env, jobject thiz, jint catId, jlong bmkId)
+{
+  Bookmark const * bmk = getBookmark(catId, bmkId);
+  search::AddressInfo const address = g_framework->NativeFramework()->GetAddressInfoAtPoint(MercatorBounds::FromLatLon(bmk->GetLatLon()));
```

Comment:
Это и есть адрес, который показывается у букмарок в плейспейдже. Просто при создании букмарки мы не заполняем адрес сразу же - иначе список букмарок при скролле серьёзно тормозит на операции GetAddressInfoAtPoint. Таким образом обеспечивается ленивость запроса адреса.

> On Mar 25, 2016, at 3:31 PM, Alex Zolotarev <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/bookmarks/data/Bookmark.cpp <https://github.com/mapsme/omim/pull/2526#discussion_r57440227>:
> 
> > @@ -81,4 +81,13 @@ Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeEncode2Ge0Url(
> >  {
> >    return jni::ToJavaString(env, frm()->CodeGe0url(getBookmark(cat, bmk), addName));
> >  }
> > +
> > +JNIEXPORT jstring JNICALL
> > +Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeGetAddress(
> > +     JNIEnv * env, jobject thiz, jint catId, jlong bmkId)
> > +{
> > +  Bookmark const * bmk = getBookmark(catId, bmkId);
> > +  search::AddressInfo const address = g_framework->NativeFramework()->GetAddressInfoAtPoint(MercatorBounds::FromLatLon(bmk->GetLatLon()));
> Погоди, тогда это не будет консистентно с тем адресом, который печатается в плейспейдже. Вопрос: зачем вообще у букмарок метод получения адреса? Какое он имеет к ним отношение? Под айфоном, например, такого вообще нет.
> … <x-msg://4/#>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> 
>  <https://github.com/mapsme/omim/pull/2526#discussion_r57439562> <https://github.com/mapsme/omim/pull/2526/files/c026c407dde3767df58339f1a8a905cbb630eeba#r57439562>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2526/files/c026c407dde3767df58339f1a8a905cbb630eeba#r57440227>